dashboard/app/models/user.rb

```diff
@@ -486,10 +486,12 @@ def email_and_hashed_email_must_be_unique
     # skip the db lookup if we are already invalid
     return unless errors.blank?
 
-    if ((email.present? && (other_user = User.find_by_email_or_hashed_email(email))) ||
-        (hashed_email.present? && (other_user = User.find_by_hashed_email(hashed_email)))) &&
-        other_user != self
+    if email.present? && other_user != User.find_by_email_or_hashed_email(email) &&
+      self != other_user
       errors.add :email, I18n.t('errors.messages.taken')
+    elsif hashed_email.present? && other_user != User.find_by_hashed_email(hashed_email) &&
+      self != other_user
+      errors.add :hashed_email, I18n.t('errors.messages.taken')
```

Comment:
I'd say not desired. I'm not even sure the error would show up since `hashed_email` is a [hidden](https://github.com/code-dot-org/code-dot-org/blob/cd5b60f1a6ea8f6a9cecef5db5cfd890949a3465/dashboard/app/views/devise/registrations/edit.html.haml#L8-L7) field.

I'd say for clarity to the user (even though less accurate), we should put the error on `email` in either case. At least verify this looks reasonable in the user account edit page.